shared/templates/mount_option_home/oval.template

```diff
@@ -0,0 +1,75 @@
+{{% set non_allowed_partitions = ["/lib",
+                                  "/opt",
+                                  "/usr",
+                                  "/bin",
+                                  "/sbin",
+                                  "/boot",
+                                  "/dev",
+                                  "/proc"] %}}
+<def-group>
+    <definition class="compliance" id="{{{ _RULE_ID }}}" version="1">
+        {{{ oval_metadata("Home mount points should be mounted with mount option " + MOUNTOPTION + ".") }}}
+        <criteria>
+            <criterion comment="{{{ MOUNTOPTION }}} on Home mount points"
+            test_ref="test_home_partition_{{{ MOUNTOPTION }}}" />
+            <criterion comment="Home directory is not in a unexpected partition"
+            test_ref="test_home_allowed_partition_{{{ MOUNTOPTION }}}" />
+        </criteria>
+    </definition>
+
+    <ind:textfilecontent54_object id="object_home_partitions_mount_points_{{{ MOUNTOPTION }}}" version="1">
+        <ind:filepath>/etc/passwd</ind:filepath>
+        <ind:pattern operation="pattern match" var_ref="var_home_partitions_mount_points_{{{ MOUNTOPTION }}}_regex"
+        var_check="at least one"/>
+        <ind:instance datatype="int">1</ind:instance>
+    </ind:textfilecontent54_object>
+
+    <linux:partition_object id="object_home_partition_all_partitions_{{{ MOUNTOPTION }}}" version="1">
+        <linux:mount_point operation="not equal">/</linux:mount_point>
+    </linux:partition_object>
+
+    <local_variable id="var_home_partitions_mount_points_{{{ MOUNTOPTION }}}_regex" datatype="string" version="1"
+    comment="Variable including all home dirs from interactive users">
+        <concat>
+            <literal_component>^(?:(?!nobody).*)?:.*?:[1-9]\d{3,}:.*?:.*?:(</literal_component>
+            <object_component item_field="mount_point"
+            object_ref="object_home_partition_all_partitions_{{{ MOUNTOPTION }}}" />
+            <literal_component>).*?:.*</literal_component>
+        </concat>
+    </local_variable>
+
+    <local_variable id="var_home_partitions_mount_points_{{{ MOUNTOPTION }}}" datatype="string" version="1"
+    comment="This variable collects all partition mount points for interactive users">
+        <object_component item_field="subexpression"
+        object_ref="object_home_partitions_mount_points_{{{ MOUNTOPTION }}}" />
+    </local_variable>
```

Comment:
The idea used here was to collect all partitions and then filter which ones are related to interactive users. I would suggest a different approach of identifying the interactive users first and then checking the respective partitions, as already used in other similar rules, like here:
https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/accounts/accounts-session/accounts_user_interactive_home_directory_exists/oval/shared.xml

This suggested approach is less complex and more flexible since the range of uids reserved for interactive users can change in different distros. Also, considering servers, it seems more likely to have less interactive users than partitions.servers, it seems more likely to have less interactive users than partitions. Therefore, it is expected this suggested approach would improve a little the template performance.

Reply:
I also thought that your suggestion was a better approach. But I just found several ways on how it doesn't work.

My issue, basically was that that I didn't found a way to filter based on "is substring of". That way I could only include those partitions which their mount point is substring of any home directory

Reply:
I see. So, basically, the intention is:
1 - Collect a list home directories owned by interactive users
2 - Collect a list of partitions mount points
3 - Filter the mount points list to contain only mount points used for home directories of interactive users
4 - Check the mount options for the filtered list of mount points

I have checked the OVAL documentation.
It is possible to use `password_object` to collect the home directories and satisfy the `Step 1`, as already mentioned. Example:
https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/accounts/accounts-session/accounts_user_interactive_home_directory_exists/oval/shared.xml#L12-L29

The `Step 2` seems good as already defined in this PR.

The `Step 3` is the challenge. I believe the OVAL operation `subset of` could solve the problem. However, I have never tried it and some tests would be necessary to confirm. I think it is worth to test since it would permit a simpler solution if it works:
https://oval-community-guidelines.readthedocs.io/en/latest/oval-schema-documentation/oval-common-schema.html#operationenumeration-1

The `Step 4` seems good as already defined in this PR.

Reply:
I think it is not possible to use `subset of` as th OVAL documentation says here https://oval-community-guidelines.readthedocs.io/en/latest/oval-schema-documentation/oval-common-schema.html#simpledatatypeenumeration-1

> The string datatype describes standard string data. This datatype conforms to the W3C Recommendation for string data. Expected operations within OVAL for string values are ‘equals’, ‘not equal’, ‘case insensitive equals’, ‘case insensitive not equal’, ‘pattern match’.

I also tried to use it and didn't work, I was able to build ol8 product with this operation applied to a string, so apparently it is syntactically correct, but it doesn't behave as expected

Reply:
Thanks for sharing this experience with the `subset of`. I sincerely didn't have time to test it and didn't find any implementation of it. It is possible the scanner (openscap) would need some adjustments to align the expected behavior, but I am also not sure about this at the moment.

In this case, your approach of filtering a `textfilecontent54_object` object seems the most reasonable for the moment.

@yuumasato , what do you think? Do you know any OVAL trick worth to try?